app/lib/services/import.dart

```diff
@@ -517,6 +583,47 @@ class ImportService {
   }
 
   NoteData? _submit(
+    NoteData document, {
+    required List<List<PadElement>> batchesOfElements,
+    List<DocumentPage> pages = const [],
+    List<Area> areas = const [],
+    bool choosePosition = false,
+    required List<PadElement> elements,
+  }) {
+    final state = _getState();
+    DocumentPage page =
+        state?.page ?? document.getPage() ?? DocumentDefaults.createPage();
+    //cicle
+
+    if (choosePosition && state != null) {
+      for (var batch in batchesOfElements) {
+        if (batch.isNotEmpty || areas.isNotEmpty) {
+          state.currentIndexCubit.changeTemporaryHandler(
+              bloc!, ImportTool(elements: batch, areas: areas));
+        }
+        var newContent = List<PadElement>.from(page.content)..addAll(batch);
+        page = page.copyWith(content: newContent);
+      }
+    } else {
+      for (var batch in batchesOfElements) {
+        bloc
+          ?..add(ElementsCreated(batch))
+          ..add(AreasCreated(areas));
+        var newContent = List<PadElement>.from(page.content)..addAll(batch);
+        page = page.copyWith(content: newContent);
+      }
+    }
+    document = document.setPage(page);
+
+    for (final page in pages) {
+      bloc?.add(PageAdded(null, page));
+      (document, _) = document.addPage(page);
+    }
+
+    return document;
+  }
+
+  NoteData? _submitNoBatch(
```

Comment:
I would call this _submit and the other _submitBatched so you don't need to change the other import methods